osu.Game/Database/DatabaseWorkingBeatmap.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.IO;
+using osu.Framework.Audio.Track;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.Formats;
+using osu.Game.Beatmaps.IO;
+
+namespace osu.Game.Database
+{
+    public partial class BeatmapDatabase
+    {
+        private class DatabaseWorkingBeatmap : WorkingBeatmap
+        {

```

Comment:
So would you prefer it become a public class? Or just internal? I'm not to thrilled about internal in this context